### ISSUE TITLE
Per-project definition file?

### ISSUE THREAD

**Issue by jcassee** [etienne-monier/LedgerTools#6]
From the documentation it seems that the plugin supports only one definition file per user. Is it possible to set a project-specific definition file in some way?

**Comment by etienne-monier** [comment]
Sublime Text support system-wide as project-wide settings. For project-wide settings, refer to [this documentation](https://www.sublimetext.com/docs/projects.html#settings-key).

Just specify project-specific settings in `.sublime-project` file.

**Comment by jcassee** [comment]
Sorry to be a bother, but setting it in the project does not seem to work:

```
{
	"folders": [
		{
			"name": "My Ledger",
			"path": "/path/to/files"
		}
	],
	"settings": {
		"definition_filename": "/path/to/files/definition.ledger"
	}
}
```

I still get the error "Ledger definition filename does not exists."

Weirdly, I tried to debug in the console:
```
>>> os.path.exists(view.settings().get("definition_filename"))
True
```

So maybe something else is going on. Any idea how to debug further?

**Comment by jcassee** [comment]
I extracted the LedgerTools package and added some debugging to `get_definition_filename`. Unfortunately, the plugin is not picking up the project settings, `definition_filename` is taken from the user settings.

I added the following to `get_definition_filename` in `util.py`:
```python
    # Check for filename in the context of the current file if it is not found
    if not os.path.exists(definition_filename):
        current_file = sublime.active_window().active_view().file_name()
        current_dir = os.path.dirname(current_file)
        definition_filename = os.path.join(current_dir, definition_filename)
```

@etienne-monier can we reopen the issue?

**Comment by ShrykeWindgrace** [comment]
I am experiencing the same issue. +1 for reopening=)

**Comment by etienne-monier** [comment]
Ok, I'll have a look on that.

**Comment by ShrykeWindgrace** [comment]
On the first sight, this is not immediately possible: per the documentation (https://www.sublimetext.com/docs/projects.html#settings-key)

> Note that only settings in the category [Editor Settings](https://www.sublimetext.com/docs/settings.html#editor-settings) may be controlled by a project.

These [editor settings](https://www.sublimetext.com/docs/settings.html#editor-settings) seem to not include plugin-specific things. I'll take a look whether we can override *syntax-specific* settings on a project basis...

**Comment by jcassee** [comment]
Indeed, that is why the code snippet above allows the `definition_filename` to be relative to the current file location. But maybe having it relative to the project file would be even better.

**Comment by ShrykeWindgrace** [comment]
Hello! Just a friendly reminder that there is PR addressing this issue =)

**Comment by etienne-monier** [comment]
Hi,
I'm currently making a whole refactor to make clear code and tests.

Concerning the settings, the keys are going to be configurable via the project settings file or the pluggin setting files.

**Comment by ShrykeWindgrace** [comment]
Thanks for the update!

**Comment by ShrykeWindgrace** [comment]
Hi! Any progress on this?
